cmd/agent/main.go

```diff
@@ -209,6 +232,12 @@ func newGrpcProxyAgentOptions() *GrpcProxyAgentOptions {
 		probeInterval:           1 * time.Second,
 		syncIntervalCap:         10 * time.Second,
 		serviceAccountTokenPath: "",
+		kubeconfigPath:          "",
+		kubeconfigQPS:           0,
+		kubeconfigBurst:         0,
+		enablePodCondition:      false,
+		podNamespace:            os.Getenv("POD_NAMESPACE"),
```

Comment:
Env vars instead of flags for easy use with fieldRef values.

Reply:
I have no issue with initializing podNamespace and podName using os.Getenv. I'm not certain these environment variables will actually be set in all environments. As such we should still have command line flags to allow for over-rides if needed.